text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
+
+**Environment variables**:
+
+* `vars` (renamed from `env`): yields a vector of `(OsStrBuf, OsStrBuf)` pairs.
+* `var` (renamed from `getenv`): take a value bounded by `IntoOsStrBuf`,
+  allowing Rust strings and slices to be ergonomically passed in. Yields an `Option<OsStrBuf>`.
+* `set_var` (renamed from `setenv`): takes two `IntoOsStrBuf`-bounded values.
+* `remove_var` (renamed from `unsetenv`): takes a `IntoOsStrBuf`-bounded value.
+
+* `join_paths`: take an `IntoIterator<T>` where `T: IntoOsStrBuf`, yield a `Result<OsStrBuf, JoinPathsError>`.
+* `split_paths` take a `IntoOsStrBuf`, yield an `Iterator<Path>`.
+
+**Working directory**:
+
+* `current_dir` (renamed from `getcwd`): yields a `PathBuf`.
+* `set_current_dir` (renamed from `change_dir`): takes an `AsPath` value.
+
+**Important locations**:
+
+* `home_dir` (renamed from `homedir`): returns home directory as a `PathBuf`
+* `temp_dir` (renamed from `tmpdir`): returns a temporary directly as a `PathBuf`
+* `current_exe` (renamed from `self_exe_name`): returns the full path
+  to the current binary as a `PathBuf`.
+
+**Exit status**:
+
+* `get_exit_status` and `set_exit_status` stay as they are, but with
+  updated docs that reflect that these only affect the return value of
+  `std::rt::start`.
+
+**Architecture information**:
+
+* `num_cpus`, `page_size`: stay as they are
```

Comment:
@tshepang you cannot hot-plug cores, only chips. Otherwise the line between physical cores and chips is very thin. What’s the meaning of CPU is mostly irrelevant to the discussion, though, since we only have one function to rename and it returns the count of logical cores.